src/sentry/preprod/eap/write.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+import uuid
+from typing import Any
+
+from arroyo import Topic as ArroyoTopic
+from arroyo.backends.kafka import KafkaPayload
+from google.protobuf.timestamp_pb2 import Timestamp
+from sentry_protos.snuba.v1.request_common_pb2 import TraceItemType
+from sentry_protos.snuba.v1.trace_item_pb2 import TraceItem as EAPTraceItem
+
+from sentry.conf.types.kafka_definition import Topic
+from sentry.preprod.eap.constants import PREPROD_NAMESPACE
+from sentry.preprod.models import PreprodArtifactSizeMetrics
+from sentry.replays.lib.kafka import EAP_ITEMS_CODEC, eap_producer
+from sentry.search.eap.rpc_utils import anyvalue
+from sentry.utils.kafka_config import get_topic_definition
+
+
+def write_preprod_size_metric_to_eap(
+    size_metric: PreprodArtifactSizeMetrics,
+    organization_id: int,
+    project_id: int,
+) -> None:
+    """
+    Write a PreprodArtifactSizeMetrics to EAP as a TRACE_ITEM_TYPE_PREPROD trace item.
+
+    NOTE: EAP is append-only, so this function should only be called after the size metric has been successfully committed
+    since we cannot update fields later on.
+    """
+    proto_timestamp = Timestamp()
+    proto_timestamp.FromDatetime(size_metric.date_added)
+
+    received = Timestamp()
+    received.FromDatetime(size_metric.date_added)
+
+    artifact = size_metric.preprod_artifact
+
+    # Generate a unique trace_id for this preprod artifact using UUID5 with PREPROD_NAMESPACE.
+    # This ensures no collision with other trace types in EAP.
+    # Design: Use preprod_artifact_id to group related components of the SAME build
+    # (e.g., main app + Watch extension + dynamic features) under one trace.
+    trace_id = uuid.uuid5(PREPROD_NAMESPACE, str(size_metric.preprod_artifact_id)).hex
```

Comment:
Does this look good? I see uptime does a similar pattern to ensure no collisions with other traces.